15-connect-app-and-store/README.md

```diff
@@ -0,0 +1,527 @@
+
+    return (
+      <main className="app">
+        <div className="app__page">
+          <div className="app__list">
+            <EmailList
+              emails={emails}
+              onItemSelect={this._handleItemSelect.bind(this)}
+              onItemDelete={this._handleItemDelete.bind(this)}
+              onItemMarkUnread={this._handleItemMarkUnread.bind(this)}
+              selectedEmailId={selectedEmailId}
+            />
+          </div>
+          <EmailViewWrapper
+            selectedEmail={selectedEmail}
+            onClose={this._handleEmailViewClose.bind(this)}
+            onDelete={this._handleItemDelete.bind(this, selectedEmailId)}
+            onMarkUnread={this._handleItemMarkUnread.bind(
+              this,
+              selectedEmailId
+            )}
+            onMarkRead={this._handleItemMarkRead.bind(this, selectedEmailId)}
+          />
+          <button
+            className="app__new-email"
+            onClick={this._handleShowForm.bind(this)}
+          >
+            +
+          </button>
+          <EmailFormWrapper
+            showForm={showForm}
+            onSubmit={this._handleFormSubmit.bind(this)}
+            onCancel={this._handleHideForm.bind(this)}
+          />
+        </div>
+      </main>
+    );
+  }
+}
+```
+
+and `App.js` should look like:
+
+```js
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import {getEmails} from './action-reducers';
+
+export default class App extends PureComponent {
+  static propTypes = {
+    pollInterval: PropTypes.number
+  };
+
+  static defaultProps = {
+    // default the `pollInterval` prop to 2 secs when not specified
+    pollInterval: 2000
+  };
+
+  componentDidMount() {
+    // Retrieve emails from server once we know DOM exists
+    this._getUpdateEmails();
+
+    // Set up long-polling to continuously get new data
+    this._pollId = setInterval(
+      () => this._getUpdateEmails(),
+      this.props.pollInterval
+    );
+  }
+
+  componentWillUnmount() {
+    // Need to remember to clearInterval when the component gets
+    // removed from the DOM, otherwise the interval will keep going
+    // forever and leak memory
+    clearInterval(this._pollId);
+  }
+
+  _getUpdateEmails() {
+    return getEmails().then(emails => this.setState({emails}));
+  }
+
+  render() {
+    return (
+    );
+  }
+}
+```
+
+Now that we have separated out `<App/>` into two distinct components, lets start by setting up `<App/>` to create a `store` and have `<Provider/>` consume the store as the top level child. Then, we will return to `<Page/>` and complete it's refactor.
+
+Next, lets create our `store`.
+
+We are going to do this in `<App/>`'s [`constructor()`](https://facebook.github.io/react/docs/react-component.html#constructor). This should be added directly after `static defaultProps` as it is the first thing called when the component is evaluated.
```

Comment:
Since this is more general Redux, I'd prefer to define the `store` in a `const` like "normal." We do it in the constructor because of server-side rendering and stuff bleeding across requests